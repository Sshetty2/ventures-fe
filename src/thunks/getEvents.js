import { setEvents, setLoading, setError } from '../actions';
import { fetchData } from '../utils/api';

export const getEvents = (event_type = '', month = '', state = '') => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      let path = (event_type.length || state.length || month.length) ? '?' : '';
      if (event_type.length) path = path + `&event_type=${event_type}`;
      if (month.length) path = path + `&month=${month}`;
      if (state.length) path = path + `&state=${state}`;
      const events = await fetchData('/events' + path, 'GET');
      dispatch(setEvents(events.data));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false))
  }
}