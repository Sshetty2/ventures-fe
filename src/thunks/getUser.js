import { setLoading, setError, setUser } from '../actions';
import { fetchData } from '../utils/api';

export const getUser = (userId = '') => { 
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const user = await fetchData(`/users/${userId}`, 'GET');
      dispatch(setUser(user.data.attributes));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}