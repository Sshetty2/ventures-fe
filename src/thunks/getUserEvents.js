import { setLoading, setError, setUserEvents } from '../actions';
import { fetchData } from '../utils/api';

export const getUserEvents = (userId) => { 
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const userEvents = await fetchData(`/users/${userId}/events`, 'GET');
      dispatch(setUserEvents(userEvents.data));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}