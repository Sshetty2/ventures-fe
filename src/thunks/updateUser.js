import { setLoading, setError, setUser } from '../actions';
import { fetchData } from '../utils/api';

export const updateUser = (user) => {
  return async (dispatch) => {
    const { name, bio } = user;
    try {
      dispatch(setLoading(true));
      const updatedUser = await fetchData(`/users/1?name=${name}&bio=${bio}`, 'PUT');
      dispatch(setUser(updatedUser.data.attributes));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}