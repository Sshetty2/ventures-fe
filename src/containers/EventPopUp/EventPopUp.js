import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getEvent } from '../../thunks/getEvent';

export const EventPopUp = ({ getEvent, match, history }) => {
  const [event, updateEvent] =  useState({});
  const [showVideo, updateShowVideo] = useState(false);

  useEffect(() => {
    const fetchEvent = async () => {
      const event = await getEvent(match.params.id);
      updateEvent(event)
    }
    fetchEvent();
  }, []);

  const displayToShow = () => {
    const { image_url, video_url } = event;
    if (showVideo) {
      return (<iframe title='Event Video' className='image-container' src={video_url}></iframe>)
    } else {
      return (<div className='image-container' style={{ backgroundImage: `url(${image_url})` }}></div>)
    }
  }

  if (event.name) {
    const { name, city, state, event_type, price, start_date, end_date, description, event_url } = event;
    return (
      <div className='overlay'>
        <div className='EventPopUp'>
          <div className='display-panel'>
            {displayToShow()}
            <button className={showVideo ? 'back arrow' : 'forward arrow'} onClick={() => updateShowVideo(!showVideo)}></button>
          </div>
          <div className='text-container'>
            <button onClick={() => history.goBack()} className='close-pop-up'></button>
            <h2>{name}</h2>
            <h4>{city}, {state}</h4>
            <h4>Dates: <span className='light-text'>
              {start_date === end_date ? `${start_date}` : `${start_date} to ${end_date}`}
            </span></h4>
            <h4 className='event-type'>Event Type: <span className='light-text'>{event_type}</span></h4>
            <h4>Price: <span className='light-text'>${price}</span></h4>
            <p>{description}</p>
            <a className='event-link' href={event_url}>Event Website</a>
            <div className='button-div'>
              <button className='wishlist-button'></button>
              <button className='attending-button'></button>
            </div>
          </div>
        </div>
      </div>
    )
  } else {
    return(<div>Loading...</div>)
  }
}

export const mapDispatchToProps = (dispatch) => ({
  getEvent: (id) => dispatch(getEvent(id)),
});

export default connect(null, mapDispatchToProps)(EventPopUp);

EventPopUp.propTypes = {
  getEvent: PropTypes.func,
  match: PropTypes.object,
  history: PropTypes.object
}