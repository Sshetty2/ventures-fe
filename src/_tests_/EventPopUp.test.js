import React from 'react';
import { shallow } from 'enzyme';
import { EventPopUp, mapDispatchToProps } from '../containers/EventPopUp/EventPopUp';
import { getEvent } from '../thunks/getEvent';

const mockProps = {
  getEvent: jest.fn()
}

describe('EventPopUp', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<EventPopUp {...mockProps} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    it.skip('should call dispatch with the getEvent thunk', async () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getEvent('1');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getEvent('1');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});