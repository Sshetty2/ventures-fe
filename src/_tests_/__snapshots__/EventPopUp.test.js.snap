// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventPopUp should match the snapshot 1`] = `
<div>
  Loading...
</div>
`;
